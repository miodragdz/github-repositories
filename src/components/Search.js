import React, { Component } from "react";

export class Search extends Component {
  state = {
    searchTerm: '',
  };

  onHandleChange = e => {
    this.setState({
      searchTerm: e.target.value
    });
  }

  onHandleSubmit = e => {
    const { onSearchButtonClick } = this.props;
    const { searchTerm } = this.state;
    e.preventDefault();
    onSearchButtonClick(searchTerm);
  }

  render() {   
    const { searchTerm } = this.state;

    return (
      <div className="search">
        <form className="form">
          <input
            id="location"
            value={searchTerm}
            placeholder="Search by name"
            onChange={this.onHandleChange}
            className="form__input"
          />
          <button className="form__button" disabled={!searchTerm} onClick={this.onHandleSubmit} type="submit">Search</button>        
        </form>
      </div>
    );
  }
}

export default Search;
