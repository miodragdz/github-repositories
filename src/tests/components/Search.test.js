import React from "react";
import { shallow } from "enzyme";
import Search from "../../components/Search";

let onSearchButtonClick, wrapper;

beforeEach(() => {
  onSearchButtonClick = jest.fn();
  wrapper = shallow(<Search onSearchButtonClick={onSearchButtonClick} />);
});

test("should correctly render Search", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should set searchTerm on input change", () => {
  const value = "New searchTerm value";
  wrapper.find("input").simulate("change", {
    target: { value },
  });
  expect(wrapper.state("searchTerm")).toBe(value);
});

test("should call onSearchButtonClick prop for form submission", () => {
  wrapper.find("button").simulate("click", {
    preventDefault: () => {},
  });
  expect(onSearchButtonClick).toHaveBeenCalledWith(wrapper.state("searchTerm"));
});
