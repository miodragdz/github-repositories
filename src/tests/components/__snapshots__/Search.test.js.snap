// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Search 1`] = `ShallowWrapper {}`;
